gufe/components/proteincomponent.py

```diff
@@ -1,86 +1,671 @@
+                    val = getattr(atom_monomer_info, f)()
+                    mi_dict[f.replace("Get", "")] = val
+
+            atoms.append(
+                (
+                    atom.GetAtomicNum(),
+                    name,
+                    atom.GetFormalCharge(),
+                    atom.GetIsAromatic(),
+                    "",  # Stereocent in Smallcomponent
+                    atom.GetPropsAsDict(),
+                    mi_dict,
+                )
+            )
+
+        bonds = [
+            (
+                bond.GetBeginAtomIdx(),
+                bond.GetEndAtomIdx(),
+                bond.GetBondType(),
+                bond.GetIsAromatic(),
+                bond.GetStereo() or "",
+                bond.GetPropsAsDict(),
+            )
+            for bond in self._rdkit.GetBonds()
+        ]
+
+        conformers = [
+            serialize_numpy(conf.GetPositions())  # .m_as(unit.angstrom)
+            for conf in self._rdkit.GetConformers()
+        ]
+
+        # Additional Information for the mol:
+        molecule_props = {}
+        for prop_key, prop_value in self._rdkit.GetPropsAsDict(
+            includePrivate=True
+        ).items():
+            if prop_key == "sequence":
+                residue_sequence = prop_value.split()
+                molecule_props["sequence"] = residue_sequence
+            elif isinstance(prop_value, str) and prop_value.startswith("{"):
+                val = json.loads(prop_value.replace("'", '"'))
+                molecule_props[prop_key] = val
+            elif isinstance(prop_value, str) and prop_value.startswith("["):
+                val = ast.literal_eval(prop_value)
+                molecule_props[prop_key] = val
+            else:
+                molecule_props[prop_key] = prop_value
+
+        # Result
+        d = {
+            "atoms": atoms,
+            "bonds": bonds,
+            "name": self.name,
+            "conformers": conformers,
+            "molecules": molecule_props,
+        }
+
+        return d
+
+    # NOT implemented:
+    def to_openeye(self) -> OEMol:
+        """OEChem representation of this molecule"""
+        raise NotImplementedError
+
+    def to_openff(self):
+        raise NotImplementedError
+
+    def to_gmx(self, out_gmx_top: str, out_gmx_gro: str):
+        raise NotImplementedError
+
+    def to_amber(self, out_path: str):
+        raise NotImplementedError
+
+    @classmethod
+    def from_openeye(cls, oemol: OEMol, name: str = ""):
+        raise NotImplementedError
+
+    @classmethod
+    def from_openff(cls, openff: OFFMolecule, name: str = ""):
+        raise NotImplementedError
 
     @classmethod
-    def _from_dict(cls, d: dict):
-        raise NotImplementedError()
+    def from_gmx(cls, gmx_top: str, gmx_gro: str):
+        raise NotImplementedError
 
-    @property
-    def total_charge(self):
-        return None
+    @classmethod
+    def from_amber(cls, name=""):
+        raise NotImplementedError
```

Comment:
is it worth even putting the skeletons of these in now? are we even planning to support on-ramp from a .gro of a protein?

Reply:
na, I will remove them.